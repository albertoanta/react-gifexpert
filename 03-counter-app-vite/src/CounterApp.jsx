// Tarea 
// https://gist.github.com/Klerith/e1a731cc595c00a9794a709062eae757


import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ( {value} ) => {
  
  console.log('------ Componente CounterApp')
  console.log ( value);
  
  const[ counter, setCounter ] = useState( value );


  const HandleAdd = ( event ) => { 
    //console.log(event)
    // console.log(+1);
    setCounter (counter +1 ) ;
   //  setCounter( (c) => c+1 ) ; // es analogo a lo anterior
  
  }

  const HandleSubstract = () => setCounter (counter -1 ) ;
   
  const HandleReset = (event) => { 
  
    setCounter (value) ;
   //  setCounter( (c) => c+1 ) ;
  
  }

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{value}</h2>
      <h4> Contador hooks {counter} </h4>

      {/* <button onClick={ (event) => HandleAdd(event) }> */}
      <button onClick={ HandleAdd }>
        +1
      </button>

      <button onClick={ HandleSubstract }> -1 </button>
      <button aria-label="btn-reset" onClick={ HandleReset }> Reset </button>
    </>
  )
} 

CounterApp.propTypes = { 
  value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
  value :1234

}