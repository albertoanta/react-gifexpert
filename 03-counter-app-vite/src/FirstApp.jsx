// import { Fragment } from "react";
// export const FirsApp = () => {
//   return (
//     <Fragment>
//       <h1> First App</h1>
//       <p>Esto es un subtitulo</p>
//     </Fragment>
//   )
// }

// En vez de importar y ponerlo como un fragmento con la etioqueta <Fragment></Fragment>
// podemos poner <> que es anÃ¡logo

const newMessage = [1,2,3,4,5,6,7,8,9];
const secondMessage = {
  message : 'Hola Mundo',
  title : 'Alberto'
  };

const  hazSaludo = () =><h3>Esto es una peticion de saludo</h3>;

import PropTypes from "prop-types";

export const FirstApp = ( {title, subTitle, name } ) => {
  

  console.log ( title);


  return (
    <>
      {/* <h1> {title} </h1> */}
      <h1 data-testid="test-title" > {title} </h1>
      {/* <h1> { `Primera App` } { newMessage}</h1> */}
      {/* <code> { JSON.stringify( secondMessage) }</code> */}
      <p>Esto es un subtitulo</p>
      { hazSaludo() }
      <p>{ subTitle +1}</p>
      <p>{ subTitle }</p>
      
      <p>{name}</p>
    </>
  )
} 

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  // subTitle: PropTypes.number
  subTitle: PropTypes.string

}

FirstApp.defaultProps = {
  name : "Alberto Anta",
  subTitle: "No hay subtitulo",
  // subTitle: 123,
  // title: "No hay titulo",

}