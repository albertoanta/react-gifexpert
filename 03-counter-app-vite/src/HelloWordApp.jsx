// function App() {
//     return (<h1>Hola Mundo!! </h1>);

// }

// export default App;

// No sgusta mas tener exportada de forma indepenciente cada funcion componente


export const HelloWordApp = () => {
  return (
    <h1>HelloWord App</h1>
  )
}

