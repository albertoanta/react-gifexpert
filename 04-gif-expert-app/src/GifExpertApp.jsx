import { useState } from "react"
import { AddCategory, GifGrid } from "./components";


export const GifExpertApp = () => {
    // const [categories, setCategories] = useState([ 'One Punch' , 'Dragon Ball']);
    const [categories, setCategories] = useState([ 'One Punch' ]);
    
    const onAddCategory = ( newCategory ) => {
        // console.log(newCategory);
        if (categories.includes(newCategory) ) return;  // Si ya existe la categoria no hace nada
        setCategories([newCategory, ...categories ]);
        // setCategories([...categories, 'Valorant']);
        // setCategories( cat => [...cat, 'Valorant']);
    }

    // console.log(categories);
    return(
       <>
       {/* titulo */}
        <h1>GifExpertApp</h1>
        
        {/* input */}
        <AddCategory
            // setCategories= {setCategories}  
            onNewCategory= { value => onAddCategory( value ) }
        />
        {/* listado de gif */}

            {categories.map( (category)  => (
                // return <li key={ category }>{ category}</li>
                    <GifGrid key={ category } category= { category } />
                   
             ))
        }
            


       </>
    )
}