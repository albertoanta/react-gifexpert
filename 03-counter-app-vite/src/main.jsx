import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import { HelloWordApp } from './HelloWordApp';


import './styles.css'




ReactDOM.createRoot(document.getElementById('root') ).render(
    <React.StrictMode>
        {/* <HelloWordApp /> */}
        {/* <FirstApp title="Hola, soy Goku" subTitle= { 123 }/> */}
        {/* <FirstApp/>  */}
        <CounterApp value= {20}/>

    </React.StrictMode>
);


