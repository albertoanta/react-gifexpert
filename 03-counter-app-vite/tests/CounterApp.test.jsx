import {  fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp";


describe('Pruebas en <CounterApp />', () => { 
    const valorInicial = 100;


    test('Debe hacer match con el snapshot', () => { 
        const { container } = render( <CounterApp value={ valorInicial} />);
        expect(container).toMatchSnapshot();
     });

     test('Debe de mostrar el valorInicial inicial de 100  ', () => { 
        
        render( <CounterApp value={ valorInicial} />);
        //screen.debug();
        expect(screen.getAllByText(valorInicial)).toBeTruthy();
      });

      test('Debe incrementar con el boton +1', () => { 
         render( <CounterApp value={ valorInicial} />);
         fireEvent.click(screen.getByText('+1'))
         expect(screen.getByRole('heading', { level :4 }).innerHTML).toContain('101');
         // expect(screen.getByText('101')).toBeTruthy();
      })

      test('Debe decrementar con el boton -1', () => { 
         render( <CounterApp value={ valorInicial} />);
         fireEvent.click(screen.getByText('-1'))
         expect(screen.getByRole('heading', { level :4 }).innerHTML).toContain('99');
         // expect(screen.getByText('99')).toBeTruthy();
      })

      test('Debe funcionar el boton resetaar a 100', () => { 
         render( <CounterApp value={ valorInicial} />);
         fireEvent.click(screen.getByText('+1'))
         fireEvent.click(screen.getByText('+1'))
         fireEvent.click(screen.getByText('Reset'));         
         expect(screen.getByText('100')).toBeTruthy();
      })


      test('Debe funcionar el boton resetaar a 100 con otra implementacion', () => { 
         render( <CounterApp value={ valorInicial} />);
         fireEvent.click(screen.getByText('+1'))
         fireEvent.click(screen.getByText('+1'))
         fireEvent.click(screen.getByRole('button',{ name: 'btn-reset'}));
         expect(screen.getByText('100')).toBeTruthy();
      })





   
    
});