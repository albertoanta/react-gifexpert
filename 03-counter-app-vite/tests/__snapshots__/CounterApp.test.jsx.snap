// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <h4>
     Contador hooks 
    100
     
  </h4>
  <button>
    +1
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
